### PR TITLE
Resolves FoldableVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the FoldableVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Generic visualizer (make a FoldableBase prototype or something with it)

**Comment by TemporalOroboros** [comment]
> Generic visualizer (make a FoldableBase prototype or something with it)

Conflicts with a generic visualizer that's already on one of the entities. Should I just copy-paste from the base prototype?

**Comment by AJCM-git** [comment]
> Conflicts with a generic visualizer that's already on one of the entities. Should I just copy-paste from the base prototype?

yeah, just add the same logic to the overwriting visualizer (make a comment like "foldable enums" or smt")

**Comment by TemporalOroboros** [comment]
> > Conflicts with a generic visualizer that's already on one of the entities. Should I just copy-paste from the base prototype?
> 
> yeah, just add the same logic to the overwriting visualizer (make a comment like "foldable enums" or smt")

Turns out all of the uses had different base states so I had to copypasta for all of them.

File: Resources/Prototypes/Entities/Structures/Furniture/rollerbeds.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 - type: entity
   id: RollerBed
-  parent: BaseItem
+  parents: BaseItem
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Structures/Furniture/rollerbeds.yml:3]: oop

**Review by AJCM-git — CHANGES_REQUESTED** [review]
Delete FoldableVisualLayers enum since its no longer used in code, dynamically generate it  in the prototype instead, also i think you can reduce boilerplate if you make another mapped layer in the sprite with a folded state that is invisible, then you can inherit the visualizer since it sets visibility and not state, for example
```yml
  - type: Sprite
    layers:
    - state: bag
      map: ["unfoldedLayer"] # dynamically generated
    - state: bag_folded 
      map: ["foldedLayer"] # dynamically generated
      visible: false
  - type: GenericVisualizer
    visuals:
      enum.FoldedVisuals.State:
        unfoldedLayer:
          True: {visible: false}
          False: {state: true}
        foldedLayer:
          True: {visible: true}
          False: {state: false}
```
(obviously you would have to copypaste for the ones that overwrite the visualizer but eh)

**Review by AJCM-git — CHANGES_REQUESTED** [review]
make a BaseFoldable prototype or something with the gen vis and inherit that whenever possible (reminder that the syntax for multiple parents is `parent: [foobar, barfoo]`

File: Resources/Prototypes/Entities/Structures/Furniture/rollerbeds.yml [hunk 2]
```diff
@@ -100,13 +120,24 @@
...
+      map: ["unfoldedLayer"]
+    - state: emergency_rollerbed_folded
+      map: ["foldedLayer"]
+      visible: false
   - type: Appearance
     visuals:
-    - type: FoldableVisualizer
-      key: emergency_rollerbed
     - type: RollerbedVisualizer
       key: emergency_rollerbed
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Structures/Furniture/rollerbeds.yml:130]: oops

File: Resources/Prototypes/Entities/Objects/Specific/Medical/morgue.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 - type: entity
   id: BodyBag_Container
+  parents: [BaseFoldable]
```
Comment (AJCM-git) [inline Resources/Prototypes/Entities/Objects/Specific/Medical/morgue.yml:3]: if you are using a single parent then declare it normally

**Review by AJCM-git — CHANGES_REQUESTED** [review]
even if an entity has multiple parents you still have to use `parent` as singular (also try running the game before pushing, many of these you can catch yourself when running)

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**